### PR TITLE
WIP handle bridge-over-default-nic scenario via a script

### PR DESCRIPTION
**Pull request by phoracek**
There are issues using kubernetes-nmstate to configure bridge over
the default NIC. Moreover, these issues are really difficult to debug.

This patch covers this specific scenario via a script (nmcli+ip link).

### CODE REVIEW

**Comment by phoracek** [comment]
@SchSeba @oshoval @dankenigsberg  please review this madness, still in progress though

File: build/bin/bridge-over-nic [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -xe
+
+bridge_name=$1
+nic_name=$2
+vlan_range_min=$3
+vlan_range_max=$4
+
+VLAN_RANGE="${vlan_range_min}-${vlan_range_max}"
```
Comment (oshoval) [inline build/bin/bridge-over-nic]: vlan_range_min/max should be mandatory ?
if so maybe worth to add a check like the ones below ?

**Comment by oshoval** [comment]
lgtm

File: build/bin/bridge-over-nic (hunk 1 above)
Comment (phoracek) [reply build/bin/bridge-over-nic]: good point, thanks

File: pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go [hunk 2]
```diff
@@ -145,3 +156,88 @@ func (r *ReconcileNodeNetworkStateConfiguration) Reconcile(request reconcile.Req
...
+        - name: eth0
+          vlans:
+          - vlan-range-min: 1
+            vlan-range-max: 4094
+EOF
+
+*/
+
+// TODO in order to maintain compatibility in future, trigger workaround if the requested configuration matches
+func detectBridgeWorkaround(desiredState nmstatev1alpha1.State) (string, string, string, string) {
```
Comment (dankenigsberg) [inline pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go]: I don't understand how the detection works. I expected to see here "if name == mad-workaround-policy", to make sure it does not kick in for run-of-the-mill policies.

Comment (phoracek) [reply pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go]: The issue with that would be that we would eventually need to introduce upgrade path or maintain the script forever. Instead, this specific scenario is handled by the script. Once we properly support it via nmstate, we can drop the workaround and this `desiredState` will do exactly the same job.

Comment (dankenigsberg) [reply pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go]: Ok

Comment (phoracek) [reply pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go]: BTW I have tried it on bare-metal KNI and it worked! Not as a part of kubernetes-nmstate though, just the script alone, deployed as a DaemonSet.

File: test/e2e/nodenetworkstate_test.go [hunk 3]
```diff
@@ -115,6 +141,42 @@ var _ = Describe("NodeNetworkState", func() {
 				}
 			})
 		})
+		FContext("with a linux bridge workaround (dhcp+trunk) up", func() {
```
Comment (qinqon) [inline test/e2e/nodenetworkstate_test.go]: s/FContext/Context/g

**Comment by phoracek** [comment]
This has been fixed properly and I'm glad we did it.

**Pull request closed without merging**